### ISSUE TITLE
Building without IDE - no longer possible ?

### ISSUE THREAD

**Issue by crucialfelix** [supercollider/supercollider#1788]
I'm revising the OS X build instructions. 

Just confirming: I don't think its possible to build the Cocoa IDE at all. 

Its also not currently possible to build just sclang and scsynth without also building the IDE.

**I should remove the following documentation, yes ?**

> While Qt is the default toolkit for both SC 3.6 and 3.7, it is still possible to build SC 3.6
using Cocoa and the old SC editor. This is controlled by a cmake switch for the IDE, which
implicitly toggles building the old editor:

    `-DSC_IDE=OFF`    -> old mac editor is built (by default using Qt)

This fails many times during configuration with:

CMake Error at server/plugins/CMakeLists.txt:264 (add_custom_command):
  Error evaluating generator expression:

    $<TARGET_FILE_DIR:SuperCollider>

  No target "SuperCollider"

Build fails too (because there is no Install directory ?).

Also cannot compile without Qt:

> There is also a switch to turn the use of Qt as GUI toolkit on or off. Switching Qt off
in turn switches Cocoa on. Turning Qt off implicitly prevents building the IDE, which in
turn triggers building the old editor, now using Cocoa:

    `-DSC_QT=OFF`     -> no Qt toolkit -> mac editor is built -> Cocoa is used/provided

**Comment by danstowell** [comment]
Two aspects to this:

1. The documentation is deifnitely wrong, it shouldn't say that the old editor can be built.
2. Building without IDE *is* possible, however you found a mac-only bug in that.

-DSC_IDE=OFF works just fine for me on linux, and I do make use of it. I think the error you mention is actually a bug in someone's special Mac cmake thingy. It could be that ide-less build is simply impossible on Mac? It'd be nice if ide-less Mac build was possible, but if no-one cares to put that in place then it might be better to simply hide that option from Mac users!

-DSC_QT=OFF also still has a meaningful use on linux, though if memory serves me right, it doesn't at present actually eliminate all of the Qt dependency which is a glitch that should be addressed at some point.

So in conclusion: yes remove the **Mac** documentation that you mention, but don't remove it from general docs. And someone could also tweak the main cmake files, either so that ide-less build works on mac (making a unixy build) or simply so that mac users don't even see the SC_IDE option.

**Comment by crucialfelix** [comment]
Agreed, its because scsynth and sclang were moved inside the OS X .app bundle and now they are not easily separable.

I think it should be a separate build target to just make sclang or just make scsynth.  I definitely want that for myself.

QT=OFF should be able to build sclang without Qt (which would shed about 70MB from the binary !) but it currently implies that the IDE is not being built which means it fails.

I'll update docs and file a new issue for getting the separate build target fixed.
